lib/src/rmui_bundle_suggestors/dart_script_adder.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:codemod/codemod.dart';
+import 'package:over_react_codemod/src/rmui_bundle_suggestors/html_script_adder.dart';
+import 'package:over_react_codemod/src/util.dart';
+
+import 'constants.dart';
+
+/// Suggestor that adds a [scriptToAdd] line after the last usage of a
+/// react-dart script in a Dart string literal or list of string literals.
+///
+/// Meant to be run on Dart files (use [HtmlScriptAdder] to run on HTML files).
+class DartScriptAdder extends RecursiveAstVisitor with AstVisitingSuggestor {
+  final String scriptToAdd;
+
+  /// Whether or not [scriptToAdd] is for production.
+  ///
+  /// This will determine if the script should be added to React dev or React prod js files.
+  final bool isProd;
+
+  DartScriptAdder(this.scriptToAdd, this.isProd);
+
+  @override
+  visitSimpleStringLiteral(SimpleStringLiteral node) {
+    // This value includes the quotation marks.
+    final stringValue = node.literal.lexeme;
+    final parent = node.parent;
+
+    // Do not add the script if it already exists in the string.
+    if (stringValue.contains(scriptToAdd)) return;
+
+    if (parent is VariableDeclaration) {
+      final scriptMatches =
+          RegExp(reactJsScriptPattern).allMatches(stringValue);
+
+      if (scriptMatches.isNotEmpty) {
+        final lastMatch = scriptMatches.last;
+
+        // Only add [scriptToAdd] if it has the same prod/dev status as the
+        // react-dart js [lastMatch] found.
+        final lastMatchValue = lastMatch.group(0);
+        if (lastMatchValue == null || isProd != isScriptProd(lastMatchValue))
```

Comment:
`.group(0)` is always going to be non-null; this should be
```suggestion
        final lastMatchValue = lastMatch.group(0)!;
        if (isProd != isScriptProd(lastMatchValue))
```

Same comment for the other two places where `group(0)` is used

Reply:
Addressed in 418e896af0cfbc359865b5eb8381eb1d21e9a007